pkg/cmd/cli/cmd/cancelbuild.go

```diff
@@ -105,7 +105,7 @@ func RunCancelBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 	// Create a new build with the same configuration.
 	if cmdutil.GetFlagBool(cmd, "restart") {
 		request := &buildapi.BuildRequest{
-			ObjectMeta: kapi.ObjectMeta{Name: build.Name},
+			ObjectMeta: kapi.ObjectMeta{Name: build.Name, Namespace: namespace},
 		}
 		newBuild, err := client.Builds(namespace).Clone(request)
```

Comment:
you hear that? that is the sound of a can of worms being opened... if we're *not* calling ValidNamespace on such subresources, should we be?